### PR TITLE
Attempt to fix dbjoin/CDD as repeatelement in joined group

### PR DESCRIPTION
**Pull request by jaanusnurmoja**
http://fabrikar.com/forums/index.php?threads/problem-in-multi-select-dropdown-in-cascadingdropdwon-under-repeat-group.36065/

Did I really find a solution? This time for databasejoin only. 

Generally it seems to working fine. Except 2 bugs (not sure is it caused by my fix or by something else):

1) I can't delete the single repeat group when min repeats is not set. 
When I set min repeats to 0 then I can do it (group is deleted) but I receive also a message:
Unknown column 'Array' in 'where clause' SQL=DELETE FROM chb_test_2013_repeat_checkboxes_2013 WHERE parent_id = Array

2) Another, not very important bug - in new repeat group the same checkboxes are checked as in the first group. Unchecking them, checking needed boxes and saving it - all works. 

3) NEW - test result for CDD shows that changes in databasejoin.php affected this element as well - that means it works. BUT the issue with deletion of single repeat group remains even if I set min repeats to 0 and remove data from watched element. 

---------------------------------------------------------------
TODO - same solution for fileupload element. 
First testing has shown that it ALMOST works when being in nonrepeated joined group. The parent_id value in AJAX file upload table is then the pk of joined group BUT... there is only one row althought I tried to upload multiple files and 'null' is written in file field. And the files were not uploaded at all. No idea where could I find the code to change as I did for dbjoin.

### CODE REVIEW

**Comment by jaanusnurmoja** [comment]
To make possible testing this pr in state when it was last modified, I had to commit some changes that appeared in main code very few times later, especially safeQuoteName related.

File: plugins/fabrik_element/cascadingdropdown/cascadingdropdown.php [hunk 1]
```diff
@@ -824,10 +824,26 @@ protected function buildQuery($data = array(), $incWhere = true, $opts = array()
 			{
 				foreach ($whereval as &$v)
 				{
-					$v = $db->quote($v);
-				}
 
-				$where .= count($whereval) == 0 ? '1 = -1' : $wherekey . ' IN (' . implode(',', $whereval) . ')';
+					// Jaanus: Solving bug: imploded arrays when chbx in repeated group         
+					
+					if (is_array($v))
```
Comment (pollen8) [inline plugins/fabrik_element/cascadingdropdown/cascadingdropdown.php]: no need to test for an array - safeQuote() will parse $v if its an array or a string value

File: plugins/fabrik_element/cascadingdropdown/cascadingdropdown.php [hunk 2]
```diff
@@ -824,10 +824,26 @@ protected function buildQuery($data = array(), $incWhere = true, $opts = array()
...
+					}
+					else
+					{
+						$v = FabrikString::safeQuote($v);
+					}
+				}
+      
+				// Jaanus: if count of where values is 0 or if there are no letters or numbers, only commas in imploded array
+				
+				$where .= count($whereval) == 0 || !preg_match('/\w/', implode(',', $whereval)) ? '1 = -1' : $wherekey . ' IN ' . '(' . str_replace(',,', ',\'\',', implode(',', $whereval)) . ')';
```
Comment (pollen8) [inline plugins/fabrik_element/cascadingdropdown/cascadingdropdown.php]: can be simplified - as $v will either be a string or an array which you can count, no need to do regular expression match on the imploded string

File: administrator/components/com_fabrik/models/element.php [hunk 3]
```diff
@@ -1174,13 +1185,21 @@ public function createRepeatElement($elementModel, $row)
...
 		$listModel = $elementModel->getListModel();
+		$groupModel = $elementModel->getGroupModel();
 
 		if (is_null($row))
 		{
 			$row = $elementModel->getElement();
 		}
 
-		$origTableName = $listModel->getTable()->db_table_name;
+		if ($groupModel->isJoin())
```
Comment (pollen8) [inline administrator/components/com_fabrik/models/element.php]: code style  should have a new line before the if statement

File: administrator/components/com_fabrik/models/element.php [hunk 4]
```diff
@@ -1151,12 +1152,22 @@ public function createRepeatElement($elementModel, $row)
 			$jdb->execute();
 		}
 		// Create or update fabrik join
-		$data = array('list_id' => $listModel->getTable()->id, 'element_id' => $row->id, 'join_from_table' => $listModel->getTable()->db_table_name,
+		if ($groupModel->isJoin())
```
Comment (pollen8) [inline administrator/components/com_fabrik/models/element.php]: code style  should have a new line before the if statement

File: components/com_fabrik/models/element.php [hunk 5]
```diff
@@ -4335,7 +4337,9 @@ public function getFilterQuery($key, $condition, $value, $originalValue, $type =
 				{
 					// Query the joined table concatenating into one field
 					$jointable = $this->getJoinModel()->getJoin()->table_join;
-					$pk = $this->getListModel()->getTable()->db_primary_key;
+					//$pk = $this->getListModel()->getTable()->db_primary_key;
```
Comment (pollen8) [inline components/com_fabrik/models/element.php]: code style - comments should be preceeded with a new line

**Comment by cheesegrits** [comment]
Was resubmitted as #1380 and merged there.  Closing this one.

**Pull request closed without merging**